src/interceptors/ClientRequest/NodeClientRequest.ts

```diff
@@ -267,9 +269,20 @@ export class NodeClientRequest extends ClientRequest {
           resolverResult.error
         )
 
-        // Treat thrown Responses as mocked responses.
+        // Handle thrown Response instances.
         if (resolverResult.error instanceof Response) {
-          this.respondWith(resolverResult.error)
+          // Treat thrown Response.error() as a request error.
+          if (isResponseError(resolverResult.error)) {
+            this.logger.info(
+              'received network error response, erroring request...'
+            )
+
+            this.errorWith(new TypeError('Network error'))
```

Comment:
This is the only difference with the `RequestError`. Response.error supports no error message and so it's translated to a generic `Network error` error.